core/sql/common/Ipc.h

```diff
@@ -572,7 +572,9 @@ class IpcConnection : public NABasicObject
 
   static const char *getConnectionStateString(IpcConnectionState s);
 
-  IpcConnection() : fileNumForIOCompletion_(InvalidGuaFileNumber) {}
+ IpcConnection() 
+   : fileNumForIOCompletion_(InvalidGuaFileNumber),
+    recvStreams_(NULL) {} // recvStream on C++ heap
```

Comment:
I don't think that heap is available here. I am way of making this change any bigger in scope by changing how IpcConnection objects are constructed. So this too will have to be deferred. We could make a list of all classes with NACollection data members on the system heap and then come back of them one by one.